contrib/mesos/pkg/node/node.go

```diff
@@ -36,11 +36,10 @@ const (
 )
 
 // Create creates a new node api object with the given hostname and labels
-func Create(client *client.Client, hostName string, labels map[string]string) (*api.Node, error) {
+func Create(client *client.Client, hostName string, slaveAttrLabels, annotations map[string]string) (*api.Node, error) {
```

Comment:
Hmm .. `Update` and `CreateOrUpdate` are immune against `annotations==nil`, there is no invariant to be maintained.

In case of `Create` if somebody sets `annotations=nil`, then `n.Annotations` will be nil too.
In case of `CreateOrUpdate` the `mergeMaps` function will merge existing node annotations and the given annotations and is immune against nil's too.